src/components/HabitsScreen.js

```diff
@@ -1,11 +1,172 @@
+import { useContext, useState } from "react";
+import axios from "axios";
+import styled from "styled-components";
+
+import UserContext from "../context/UserContext";
 import BottomBar from "./BottomBar";
 import TopBar from "./TopBar";
+import arrayDays from "../assets/shared/objectDays";
+import RenderHabits from "./RenderHabits";
+import AddHabit from "./AddHabit";
 
 export default function HabitsScreen() {
+    const { userInfo, habits, setHabits } = useContext(UserContext);
+    const [callHabits, setCallHabits] = useState(true);
+    const [newHabits, setNewHabits] = useState({
+        name: "",
+        days: []
+    });
+    const [blockInput, setBlockInput] = useState(false);
+    const [renderDays, setRenderDays] = useState({ ...arrayDays });
+    const [renderNewHabits, setRenderNewHabits] = useState(false);
+
+    const config = {
+        headers: {
+            "Authorization": `Bearer ${userInfo.token}`
+        }
+    }
+
+    if (callHabits) {
+        const promise = axios.get('https://mock-api.bootcamp.respondeai.com.br/api/v2/trackit/habits', config);
+        promise.then(response => {
+            setHabits(response.data);
+            setCallHabits(false);
+        });
```

Comment:
**suggestion**: I think it would be good to create a folder called service and in it contain a file called api.js, in this file all possible calls to the api would be available and you would only need to import in the files that it would return the promises. WDYT?